### ISSUE TITLE
fieldName doesn't strip JSON tag options, producing wrong error keys

### ISSUE THREAD

**Issue by cinar** [cinar/checker#194]
## Problem

`checker.go`'s `fieldName` takes the `json` tag value as-is:

```go
func fieldName(prefix string, field reflect.StructField) string {
	name := field.Name

	if jsonTag, ok := field.Tag.Lookup("json"); ok {
		name = jsonTag
	}
	...
}
```

A field tagged `json:"email,omitempty"` produces the error map key `"email,omitempty"` instead of `"email"`. A field tagged `json:"-"` produces the error key `"-"` instead of falling back to the Go field name.

`schema.go`'s `jsonPropertyName` already handles this correctly — it strips options with `strings.Cut(tag, ",")` and treats `"-"` as "no JSON name" — so `CheckStruct`'s error keys and `JSONSchema`'s property names currently disagree for any field with tag options.

## Fix

Align `fieldName` in `checker.go` with the existing logic in `schema.go`'s `jsonPropertyName` (ideally by extracting one shared helper both call), so both stripping `,omitempty`-style options and the `"-"` skip-tag case are handled consistently.

## Repro

```go
type Person struct {
	Email string `json:"email,omitempty" checkers:"required"`
}

errs, _ := checker.CheckStruct(&Person{})
// errs has key "email,omitempty" instead of "email"
```
